### PR TITLE
Final changes to make Verdi OPAM ready

### PR DESCRIPTION
**Pull request by palmskog**
Migrate all OCaml files to separate `verdi-runtime` library. Adjust `coqproject.sh` to allow both OPAM deps and manual deps. Fix installation issues. Add OPAM metadata file. Use OPAM in Travis builds. Turned off downstream Travis build for now.

### CODE REVIEW

File: opam [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+]
+install: [ make "install" ]
+remove: [ "sh" "-c" "rm -rf '%{lib}%/coq/user-contrib/Verdi'" ]
+depends: [
+  "coq" {>= "8.5" & < "8.6~"}
+  "coq-mathcomp-ssreflect" {>= "1.6"}
+  "InfSeqExt" {>= "1.0.1"}
+  "StructTact" {>= "1.0"}
+]
+tags: [ "keyword:infinite transition systems" "date:2017-01-12"]
```
Comment (dwoos) [inline opam]: Can we change this to "distributed systems" or something?

File: script/coqproject.sh [hunk 2]
```diff
@@ -24,24 +24,30 @@ function dep_dirs_lines(){
       namespace_var=${namespace_var//-/_}
       namespace_var=${namespace_var//./_}
       namespace=${!namespace_var:=$2}
-      LINE="-Q $1/$dep_dir/ $namespace"
+      if [ $dep_dir = "." ]; then
```
Comment (dwoos) [inline script/coqproject.sh]: Probably worth thinking about merging these changes into upstream `coqproject` at some point

File: opam (hunk 1 above)
Comment (palmskog) [inline opam]: Could you pick a few categories and keywords from here maybe: https://coq.inria.fr/opam/www/

Comment (dwoos) [inline opam]: None of the categories seem all that relevant, but let's go with the `distributed algorithms` keyword

File: script/coqproject.sh (hunk 2 above)
Comment (palmskog) [inline script/coqproject.sh]: Sure, but the key change is that unless you explicitly pass `$dep_PATH`, `_CoqProject` will not be changed. In other words, `../$dep` is not the default location anymore, `user-contrib' is. Do we want this for all `coqproject`-ized projects?

**Pull request merged**